packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m

```diff
@@ -819,7 +819,7 @@ + (NSDictionary *)_exportAsset:(PHAsset *)asset
     NSString *fileName = [asset valueForKey:@"filename"];
     
     return @{
-             @"id": asset.localIdentifier,
+             @"id": EXNullIfNil(asset.localIdentifier),
```

Comment:
Hmm, that should prevent an app from crashing, but I don't think it's the correct solution. Instead, you should change the `if` above to check if the `asset. localIdentifier ` is valid. I know that the `id` can be null if the asset was moved to the iCloud, but the information about this asset is still present in the local database. But there's still a small catch. Simply changing the `if` isn't enough. The app still will be crashing. To prevent that you have to filter out all the `nil` values returned from `_exportAsset`.

Reply:
> Instead, you should change the if above to check if the asset.localIdentifier is valid.

Agree.

> To prevent that you have to filter out all the nil values returned from _exportAsset.

Do you want me to check all the values used in `_exportAsset`?

Reply:
> Do you want me to check all the values used in _exportAsset?

Yes ;)

Reply:
Seeing the log closely (`objects[1]`) , I noticed this is because `filename` is nil.